SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialShadowActor.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma optimize("", off)
+#include "EngineClasses/SpatialShadowActor.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialShadowActor);
+
+USpatialShadowActor::USpatialShadowActor(const FObjectInitializer& ObjectInitializer /*= FObjectInitializer::Get()*/)
+	: Super(ObjectInitializer)
+	, EntityId(SpatialConstants::INVALID_ENTITY_ID)
+	, ReplicatedPropertyHash("")
+
+{
+}
+
+void USpatialShadowActor::Init(const Worker_EntityId InEntityId, AActor* InActor)
```

Comment:
Pass InActor as ref rather than pointer, then the assert below is unnecessary (can still store as pointer if required).

Reply:
Thanks, updated.